tests/unit/sanitizeStringForJSONParseTest.js

```diff
@@ -0,0 +1,58 @@
+import sanitizeStringForJSONParse from '../../src/libs/sanitizeStringForJSONParse';
+
+// Bad inputs should just be ignored and an empty string returned.
+const badInputs = [
+    ['', ''],
+    [null, ''],
+    [undefined, ''],
+    [42, ''],
+    [true, ''],
+];
+
+// Invalid JSON Data should be able to get parsed and the parsed result should match the input text.
+const invalidJSONData = [
+    ['Hello \t world!', 'Hello \t world!'],
+    ['Hello \n world!', 'Hello \n world!'],
+    ['Hello \n\tworld!', 'Hello \n\tworld!'],
+    ['"Hello world!"', '"Hello world!"'],
+    ['Test "', 'Test "'],
+    ['something `\\ something', 'something `\\ something'],
+
+    // Real-life examples from git commits that broke getMergeLogsAsJSON
+    // From https://github.com/Expensify/App/commit/e472470893867648cfbd85a5c2c5d24da1efece6
+    ['Add \\', 'Add \\'],
+
+    // From https://github.com/Expensify/App/pull/13500/commits/b730d5c43643f32baa3b189f0238f4de61aae0b7
+    ['Prevent commit messages that end in `\\` from breaking `getMergeLogsAsJSON()`', 'Prevent commit messages that end in `\\` from breaking `getMergeLogsAsJSON()`'],
+];
+
+// Valid JSON Data should be able to get parsed and the input text should be unmodified.
+const validJSONData = [
+    ['Hello world!', 'Hello world!'],
+    ['Hello\\\\world!', 'Hello\\\\world!'],
+];
+
+describe('santizeStringForJSONParse', () => {
+    describe.each(badInputs)('canHandleBadInputs', (input, expectedOutput) => {
+        test('sanitizeStringForJSONParse', () => {
+            const result = sanitizeStringForJSONParse(input);
+            expect(result).toStrictEqual(expectedOutput);
+        });
+    });
+
+    describe.each(invalidJSONData)('canHandleInvalidJSON', (input, expectedOutput) => {
+        test('sanitizeStringForJSONParse', () => {
+            const badJSON = `{"key": "${input}"}`;
+            expect(() => JSON.parse(badJSON)).toThrow();
+            const goodJSON = JSON.parse(`{"key": "${sanitizeStringForJSONParse(input)}"}`);
```

Comment:
NABish but I think it might make more sense to pass the whole string:

```suggestion
            const goodJSON = JSON.parse(sanitizeStringForJSONParse(badJSON));
```

Reply:
The method is not meant to sanitize the whole JSON string - just what's contained within `"subject"`

Reply:
I am confused what you mean but also this is fine so resolving